spacy_llm/tests/tasks/test_ner.py

```diff
@@ -868,15 +864,88 @@ def test_label_inconsistency():
     prompt_examples = nlp.get_pipe("llm")._task._prompt_examples
     assert len(prompt_examples) == 2
     assert prompt_examples[0].text == "Jack and Jill went up the hill."
-    assert prompt_examples[0].entities == {
-        "LOCATION": ["hill"],
-        "PERSON": ["Jack", "Jill"],
-    }
+    assert prompt_examples[0].spans == [
+        SpanReason(
+            text="Jack",
+            is_entity=True,
+            label="PERSON",
+            reason="is the name of a person",
+        ),
+        SpanReason(
+            text="Jill",
+            is_entity=True,
+            label="PERSON",
+            reason="is the name of a person",
+        ),
+        SpanReason(
+            text="went up", is_entity=False, label="==NONE==", reason="is a verb"
+        ),
+        SpanReason(
+            text="hill", is_entity=True, label="LOCATION", reason="is a location"
+        ),
+    ]
     assert (
         prompt_examples[1].text
         == "Jack and Jill went up the hill and spaCy is a great tool."
     )
-    assert prompt_examples[1].entities == {
-        "LOCATION": ["hill"],
-        "PERSON": ["Jack", "Jill"],
-    }
+    assert prompt_examples[1].spans == [
+        SpanReason(
+            text="Jack",
+            is_entity=True,
+            label="PERSON",
+            reason="is the name of a person",
+        ),
+        SpanReason(
+            text="Jill",
+            is_entity=True,
+            label="PERSON",
+            reason="is the name of a person",
+        ),
+        SpanReason(
+            text="went up", is_entity=False, label="==NONE==", reason="is a verb"
+        ),
+        SpanReason(
+            text="hill", is_entity=True, label="LOCATION", reason="is a location"
+        ),
+    ]
+
+
+@pytest.mark.parametrize(
+    "text, response, gold_ents",
+    [
+        (
+            "The woman Paris was walking around in Paris, talking to her friend Paris",
+            "1. Paris | True | PER | is the name of the woman\n"
+            "2. Paris | True | LOC | is a city in France\n"
+            "3. Paris | True | PER | is the name of the woman\n",
+            [("Paris", "PER"), ("Paris", "LOC"), ("Paris", "PER")],
+        ),
+        (
+            "Walking around Paris as a woman named Paris is quite a trip.",
+            "1. Paris | True | LOC | is a city in France\n"
+            "2. Paris | True | PER | is the name of the woman\n",
+            [("Paris", "LOC"), ("Paris", "PER")],
+        ),
+    ],
+    ids=["3_ents", "2_ents"],
+)
+def test_regression_span_task_response_parse(
+    text: str, response: str, gold_ents: List[Tuple[str, str]]
+):
+    """Test based on on spaCy issue: https://github.com/explosion/spaCy/discussions/12845
```

Comment:
```suggestion
    """Test based on spaCy issue: https://github.com/explosion/spaCy/discussions/12812
```
(which is the original issue where this example came from)